### PR TITLE
nixos/eintopf: init; eintopf: init at 0.13.16

### PR DESCRIPTION
**Pull request by onny**
## Description of changes

Packaging the community event and calendar web app [Eintopf](https://codeberg.org/Klasse-Methode/eintopf), including module for the web app server

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by onny** [comment]
```
index.html    package.json.license  src   tsconfig.json          vite.config.ts  yarn.lock.license
package.json  router.go             test  tsconfig.json.license  yarn.lock
```

```
build       node_modules  package.json.license  src   tsconfig.json          vite.config.ts  yarn.lock.license
index.html  package.json  router.go             test  tsconfig.json.license  yarn.lock
```

File: nixos/modules/services/web-apps/eintopf.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    systemd.services.eintopf = {
+      description = "Community event calendar web app";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" ];
+      wants = [ "network-online.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.eintopf}/bin/eintopf";
+        WorkingDirectory = "/var/lib/eintopf";
+        StateDirectory = "eintopf" ;
+        Environment = mapAttrsToList (k: v: "${k}=${v}") cfg.settings;
```
Comment (e1mo) [inline nixos/modules/services/web-apps/eintopf.nix:53]: Why not use `systemd.services.<name>.environment`?

**Review by e1mo — CHANGES_REQUESTED** [review]
Looks fine from going trough the nix code, except for the license being wrong ;)

**Comment by onny** [comment]
Thank you @e1mo, fixed it 👍

**Comment by e1mo** [comment]
Maybe add a note for the person doing the next update to double-check the license. They'll be switching back to AGPL (See https://codeberg.org/Klasse-Methode/eintopf/issues/351#issuecomment-2076870). However, I think that's only applicable with the next release.

**Review by e1mo — APPROVED** [review]

**Comment by onny** [comment]
Thank you both!

**Pull request merged**